### ISSUE TITLE
Integrated gradients dont match predictions

### ISSUE THREAD

**Issue by daniel-althoff** [rstudio/keras#1228]
Hi, 
I built a LSTM using 1 input variable to predict another variable as the output.
When I obtain the gradients and integrate them over the lookback series, I expected the result to be similar/equal to the network prediction. But they are somewhat different. Can anyone explain to me why?

This is how I am obtaining the gradients:
```
tensorflow::tf$compat$v1$disable_eager_execution()

input <- layer_input(shape = list(lookback, length(X)))
output <- input %>% 
  layer_lstm(units = 32, input_shape = list(lookback, length(X))) %>% 
  layer_dropout(rate = 0.2) %>% 
  layer_dense(units = length(Y))
model <- keras_model(input, output)
keras::load_model_weights_hdf5(model, './2nd_try_group_runoff_vs_precipitation/Weights_NN/Weights_test.hdf5')

gradfunc <- k_gradients(model$output, model$input)[[1]]
sess <- tf$compat$v1$keras$backend$get_session()

data_gen <- data_all_gen()
x_train <- data_gen[[1]]
y_train <- data_gen[[2]]
# Get gradient for batch
grad <- sess$run(gradfunc, feed_dict = dict(input = x_train))
```

Although there is some similarity to the prediction, the ranges are just different and overall strange.
```
model %>% predict_on_batch(x_train)
```
![image](https://user-images.githubusercontent.com/36540557/122184491-a37e1c80-ce62-11eb-8baf-03b2dce0f40a.png)

Can anyone clarify this to me?

**Comment by t-kalinowski** [comment]
Hi, thanks for filing.
This doesn't look like a bug in tensorflow or keras, but a more general DL question, so I'm going to close the issue. Stackoverflow might be a better venue for this type of question.

Some general pointers:
a) What is the loss function? I would expect gradients to be directly proportional to the loss.
b) What is the lookback length? The behavior of an LSTM is going to be sensitive to how you chose to train it, in particular the lookback length (and the loss function).
c) Is there perhaps a lag/offset between the gradient and the labels you maybe need to account for in your plot? (perhaps an offset of `lookback`)?
d) When posting for help, it's helpful to try to boil down your issue to a small reproducible example. Ideally someone should be able to copy and paste a code snippet into their terminal and reproduce the output you have questions about.
